api/openapi.yaml

```diff
@@ -2047,13 +2047,54 @@ components:
             $ref: '#/components/schemas/AssetScan'
           readOnly: true
 
-    ResourceCleanupState:
-      type: string
-      enum:
-        - Pending
-        - Skipped
-        - Failed
-        - Done
+    ResourceCleanupStatus:
+      type: object
+      required:
+        - state
+        - reason
+        - lastTransitionTime
```

Comment:
If you make these required then it is no longer possible to PATCH just one of these fields, I assume that we should always be PATCHing all of them when making a change?

Reply:
Yeah, that was our thinking here, to make sure that the whole `Status` object is updated when PATCHing. I am not sure if it make sense to update only one filed but not the others.